DOS-730 PT-Fixes CSP-Violation errors, added nonce to style tag removes violation

ajeetgill opened the pull request.

ajeetgill committed:
DOS-730 PT-Fixes CSP-Violation errors, added nonce to style tag removes violation

jlhughes commented on src/foam/core/servlet/VirtualHostRoutingServlet.js:
```diff
@@ -189,7 +189,7 @@ foam.CLASS({
       // Loading screen styles
       out.println("""
-        <style>
+        <style nonce="123455443210000">
```
What are you working around?  The nonce value should be a one time use token/value.
Recently the CSP was updated with a new library, but the sha code was not updated. 
Are you looking for something for development deployment only?

kgrgreer replied:
Yes, this should actually be one time and it should also be sent in the header. And where we add <style> elements directly we should include the nonce there as well.

jlhughes replied:
@kgrgreer was the final solution to add nonce to TP's CSP block? 
script-src 'nonce-###'
@ajeetgill 
Or, assuming you are dealing with other dynamic libraries, then you can nonce to your CSP, and then update
JsLib to take other attributes like nonce, integrity, ...  Then each script can be configured accordingly. 
see https://content-security-policy.com/nonce/ - Using nonce on external scripts. - covers the CSP and script setup.

jlhughes replied:
Forget the JsLib idea.  the CSP still needs the nonce value. 
Think you should just add a static nonce value to TP's CSP.

ajeetgill committed:
resolved minor merge conflict

ajeetgill committed:
Make CSP-nonce configurable per app via the 'http' CSpec or by reading meta tag in order to apply it consistently to server/client to avoid hardcoded duplicates.

jlhughes commented on src/foam/core/servlet/VirtualHostRoutingServlet.js:
```diff
@@ -188,8 +194,9 @@ foam.CLASS({
 
       // Loading screen styles
+      out.println("<meta name=\\"csp-nonce\\" content=\\"" + getCspNonce() + "\\">");
+      out.println("<style nonce=\\"" + getCspNonce() + "\\">");
```
This should only be output when non-null. 
The cspNonce default should be null. 
Then in the PT services.jrl you can add one and then only PT deployments will have a nonce value.

ajeetgill committed:
Make CSP nonce optional in VirtualHostRoutingServlet.
The nonce is now only included when explicitly configured, removing the default value and making CSP headers more flexible.
